docs/src/custom.md

```diff
@@ -0,0 +1,126 @@
+# Custom Transforms
+A short guide for extending the interface with your own custom transform.
+
+## Motivation
+
+TableTransforms currently supports over 25 different transforms that cover a wide variety of use cases ranging from ordinary table operations to complex statistical transformations and that all can be arbitrarily composed with one another through elegant syntax. It is easy to leverage all of those benefits as a developer of a new transforms that operates on tables so that it perfectly integrates with the rest of the TableTransforms ecosystem.
+
+##  Custom Transform Basics
+All the transforms in this package implement the transforms interface defined in the `TransformsBase` package so this is really the only dependency you need. The interface assumes the following about your implementation:
+
+- Your transform operates on a single table 
+- Your transform may be associated with some state that if computed while applying it for the first time and is then cached can help later reapply the transform on another table without recomputing the state
+- Your transform may be revertible, meaning that a transformed table can be brought back to its original form, and it may need to use the cache for that
+- Your transform may be invertible in the mathematical sense
+
+!!! note "Revertible does not imply invertible"
+    Reverting assumes that the transform has been applied already and reverts it. Meanwhile, invertible implies that there is a one-to-one mapping between the input and output tables so a table can be inverted to a corresponding input to the transform even if it was not transformed a priori.
+
+## Creating a custom transform
+In the following we shall demonstrate the steps to apply a custom transform. 
+
+### 1. Declare a new type for your transform
+The declaration struct should subtype `TransformsBase.Transforms` and it should have a named field for each input needed to apply the transform besides of the main table. For instance, if you want to call your transform `Standardize` and it takes two boolean inputs `center` and `scale`, then you should declare:
+
+```julia
+struct Standardizer <: TransformsBase.Transform
+    center::Bool
+    scale::Bool
+end
+```
+You may implement keyword instructors as needed if some of the hyperparameters are optional. Thus, we follow up with:
+```julia
+Standardize(;center::Bool=true, scale::Bool=true) = Standardize(center, scale)
+```
+
```

Comment:
```suggestion
## Defining a new transform

In the following we shall demonstrate the steps to define a new transform.

### 1. Declare a new type for your transform

The type should subtype `TransformsBase.Transform` and it should have a named field for each input needed to apply the transform besides the input table. For instance, if you want to call your transform `Standardize` and it takes two boolean inputs `center` and `scale`, then you should declare:

```julia
struct Standardize <: TransformsBase.Transform
    center::Bool
    scale::Bool
end
```

You may implement keyword constructors as needed if some of the hyperparameters are optional.
Thus, we follow up with:

```julia
Standardize(;center::Bool=true, scale::Bool=true) = Standardize(center, scale)
```

```

Reply:
I think this commit got messed up. Will fix it.